functests/execaccess_policies.py

```diff
@@ -55,6 +58,52 @@
 ]
 
 
+def poll_subject_confirmed(arch, identity):
+    """
+    polls for the subject to be confirmed for POLL_FREQUENCY every POLL_INTERVAL.
+
+    :raises: AsssertionError if the subject is not confirmed within the polling window.
+    """
+
+    for x in range(POLL_FREQUENCY):
+        subject = arch.subjects.read(identity)
+
+        if subject["confirmation_status"] == "CONFIRMED":
+            return
+
+        sleep(POLL_INTERVAL)
+
+    raise AssertionError(f"subject: {subject['identity']} is not confirmed.")
+
+
+def reciporcal_subects(arch_1, arch_2):
+    """
+    creates reciprocal subjects for arch 1 and arch 2.
+
+    Returns the id's of the imported subjects as a tuple
+    """
+    self_subject_1 = arch_1.subjects.read(SELF_SUBJECT)
+    self_subject_2 = arch_2.subjects.read(SELF_SUBJECT)
+
+    subject_1 = arch_1.subjects.create(
+        "org2_subject",
+        self_subject_2["wallet_pub_key"],
+        self_subject_2["tessera_pub_key"],
+    )
+
+    subject_2 = arch_2.subjects.create(
+        "org1_subject",
+        self_subject_1["wallet_pub_key"],
+        self_subject_1["tessera_pub_key"],
+    )
+
+    # check the subjects are confirmed
+    poll_subject_confirmed(arch_1, subject_1["identity"])
+    poll_subject_confirmed(arch_2, subject_2["identity"])
```

Comment:
This is OK but better would be to add the confirm=   keyword to the subjects.create method...

I will approve this but would you like to try that?

Reply:
yeh i will take a look :)

Reply:
due to the confirmation of subject creation needing an external call to be made using a different tenant, we will keep this as is for now, and raise a ticket for confirming subjects.

https://github.com/jitsuin-inc/archivist-python/issues/101

Reply:
Raise an issue on github